### DISCUSSION TITLE
Why is SQLite Browser trying to record screen?

### DISCUSSION THREAD

**Discussion by jonathannwinters** [sqlitebrowser/sqlitebrowser#2577] [Q&A]
Why is SQLite Browser trying to record screen?
![Screen Shot 2021-02-03 at 12 45 31 PM](https://user-images.githubusercontent.com/17182153/106922114-b45f9a00-66da-11eb-823e-3af93e6385ce.png)

**Comment by chrisjlocke** [comment]
We've not had reports of this before.  The .app suffix is also questionable, but I've not used a mac, so unsure if that's right or not.
I assume you downloaded that from our official website?
I can't think of a reason why it would 'record' the screen, but we do interrogate the screen to work out dimensions, etc, but can't imagine it's related to that...

**Reply by jonathannwinters** [reply]
Thanks for your reply. Yes downloaded from your direct web page: https://sqlitebrowser.org/dl/ 
Perhaps analyze the latest dmg just in case, and put in a query to Apple regarding your screen calls? Perhaps they may shed some light on the subject?

**Comment by chrisjlocke** [comment]
Just as a point of interest (which is a bit like my Mum looking under the bonnet of a car and asking how many horses is it...) but what operating system do you have?  We've a couple of Apple lovers who may know and have more experience in the popup that appeared.

**Comment by justinclift** [comment]
Interesting.  There should **not** be any screen recording happening, with a legit version of our software.

Would you be ok to try again, with this specific download?

&nbsp; &nbsp; https://github.com/sqlitebrowser/sqlitebrowser/releases/download/v3.12.1/DB.Browser.for.SQLite-3.12.1-v2.dmg

Also, would you be ok to verify the SHA256 checksum of it once you've downloaded it (before installing)?

The SHA256 checksum should be `45c8789dc77461299a0aeb9eaae2d089086e76eebae31b45b7a0ee48c1c9e898`.

The checksum thing is just in case something dodgy is running on your computer, changing the things you're downloading.  Hopefully that's not happening, but lets check just in case...

**Reply by jonathannwinters** [reply]
Confirmed: 45c8789dc77461299a0aeb9eaae2d089086e76eebae31b45b7a0ee48c1c9e898

Running macOS 10.15.7 on a MacBook Pro (Retina, Mid 2012)

I originally clicked DENY so the entry for Privacy  shows unchecked.


![Screen Shot 2021-02-04 at 9 54 40 PM](https://user-images.githubusercontent.com/17182153/106983250-96725380-6733-11eb-8fd6-2a01d218d15f.png)

**Reply by jonathannwinters** [reply]
Also, confirming the download file was the same one Downloaded on Feb 1: 

![Screen Shot 2021-02-04 at 9 58 00 PM](https://user-images.githubusercontent.com/17182153/106983632-36c87800-6734-11eb-95c6-8cad73fb1794.png)

**Comment by justinclift** [comment]
Cool. Well, at least it's not a case of weird, dodgy software running on your PC and changing your downloads. :smile:

Now, for the error itself... that's weird.  There's no code in DB4S for doing any screen recording, so it's most likely something just tripping up whatever Apple uses for detection.

@revolter Any idea what could cause this warning to be displayed?  Looking around online, I can see other applications have hit the same warning but there's no clear info what it's really about.

**Reply by revolter** [reply]
I found https://stackoverflow.com/a/58142253/865175, but don't know if/why/when would Qt call an API like that. I also have macOS 11.2, and that alert is not showing up for me. I think that the best bet would be for someone to ask on the Qt forums.

It's hard to manually dig this up because there isn't a specific method or key to search for in the Qt source code :(

**Comment by evdutt** [comment]
Just got this popup on 3.12.2 on MacOS 11.4. 
SHA 256 checksum of 546d57b6c88c2be7517759c016c0bf0313dfcc14adfcb43967f3c5d24657f366
Can you please verify this checksum?

Occurred as I was adding a new record to a table, and clicked away to another window on a different monitor with the "Add New Record" window still open (focused away from DB Browser for SQLite), have had the software installed and using it for about two hours now. Added a few records to another table in the same database without this popup occurring. The table I was trying to add to when the popup happened had constraints in it, but the tables that I was adding records to before the popup did not. Unsure if that provides any clues, just trying to provide more info.

**Comment by justinclift** [comment]
The SHA256 checksums for our windows and macOS releases are on the release info page for each release. :smile:

For 3.12.2, that's:

* https://sqlitebrowser.org/blog/version-3-12-2-released/, and also:
* https://github.com/sqlitebrowser/sqlitebrowser/releases/tag/v3.12.2

So, yours matches the release checksum.  No idea about the record screen thing.  I still think it'll turn out to be Qt doing something weird and/or some interaction between Qt and macOS's evolving privacy settings.

Things still work ok even without it having that permission yeah?

**Reply by evdutt** [reply]
Ah, thanks, I had only looked at the downloads page for the checksum.
Everything seems to be working fine without the permission.

**Comment by rkobylinski** [comment]
Issue still happens with v3.13.1 downloaded here: https://download.sqlitebrowser.org/DB.Browser.for.SQLite-v3.13.1.dmg

```
% echo "a641cfbfcc2ce609f07de44a35134dab53485ecc18e6d9afa297b514d74bd75e  /Users/.../Downloads/DB.Browser.for.SQLite-v3.13.1.dmg" | shasum -a 256 -c
/Users/.../Downloads/DB.Browser.for.SQLite-v3.13.1.dmg: OK
```

**Reply by justinclift** [reply]
Yeah, it's probably some Qt interaction with Apple's "Privacy" (I say that sarcastically) settings.  We don't have code in our application to do anything with screen recording.

**Reply by rkobylinski** [reply]
Whatever code it is, it's requesting to do something it should not and TBH, I am glad Apple is catching that. Maybe the log entry helps:

`informationen	11:10:48.108835+0100	tccd	Handling access request to kTCCServiceScreenCapture, from Sub:{net.sourceforge.sqlitebrowser}Resp:{TCCDProcess: identifier=net.sourceforge.sqlitebrowser, pid=56961, auid=501, euid=501, responsible_path=/Applications/DB Browser for SQLite.app/Contents/MacOS/DB Browser for SQLite, binary_path=/Applications/DB Browser for SQLite.app/Contents/MacOS/DB Browser for SQLite}, ReqResult(Auth Right: Unknown (None), promptType: 1,DB Action:None, UpdateVerifierData)`
